cdap-ui/app/cdap/components/CaskWizards/PluginArtifactUpload/UploadJsonStep/index.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { PropTypes } from 'react';
+import { connect, Provider } from 'react-redux';
+import PluginArtifactUploadStore from 'services/WizardStores/PluginArtifactUpload/PluginArtifactUploadStore';
+import PluginArtifactUploadActions from 'services/WizardStores/PluginArtifactUpload/PluginArtifactUploadActions';
+import Dropzone from 'react-dropzone';
+import T from 'i18n-react';
+import Rx from 'rx';
+
+require('./UploadJsonStep.less');
+
+const DragNDropFile = ({file, onDropHandler, error}) => {
+  return (
+    <Dropzone
+      activeClassName="file-drag-container"
+      className="file-drop-container"
+      onDrop={onDropHandler}>
+      <div className="file-metadata-container text-center">
+        {
+          file.name && file.name.length ? (<span>{file.name}</span>)
+            :
+            (
+              <span>
+                 Drag and Drop the file to be uploaded
+                <br />
+                or
+                <br />
+                Click to select file from your computer
+              </span>
+            )
+        }
+        {
+          error ?
+            <div className="text-danger">
+              {error}
+            </div>
+          :
+            null
+        }
+      </div>
+    </Dropzone>
+  );
+};
```

Comment:
Seems like we should extract this exact component, since we are reusing it in several places already?